### DISCUSSION TITLE
Tipagem de parâmetros e retorno de métodos

### DISCUSSION THREAD

**Discussion by samuelrvg** [DesignLiquido/delegua#438] [Ideias]
Definir uma forma para tipagem dos dados tanto nos parâmetros quanto dos retornos dos métodos.

Sugestões:

```
função somar(inteiro a, inteiro b) -> inteiro { .... } 
função processar(inteiro a, inteiro b) -> vazio { .... }
função mostrar(inteiro a, texto b) -> texto { .... }
```

Podemos deixar como opcional assim como no `TypeScript`. 
E caso seja definido um tipo, gerar erro caso seja passado um tipo que não corresponde ao esperado.

**Comment by leonelsanchesdasilva** [comment]
Minha ideia na verdade é:

```
função somar(a: inteiro, b, inteiro): inteiro { .... } 
função processar(a: inteiro, b: qualquer): vazio { .... }
função mostrar(a: inteiro , b: texto): texto { .... }
```

**Reply by leonelsanchesdasilva** [reply]
A avaliação de tipos, e respectivos erros, deverá ser feita na avaliação semântica.

**Reply by ItaloCobains** [reply]
@leonelsanchesdasilva @samuelrvg gostei da ideia, vai ajudar muito para escrever o tradutor para AssemblyScript.

**Comment by samuelrvg** [comment]
Com essa PR podemos encerrar esse assunto.

https://github.com/DesignLiquido/delegua/pull/458
